### PR TITLE
chore: update verification script and documentation

### PR DESCRIPTION
**Pull request by delaaxe**
- Adds support for `aws-vault` for config usage
- Better error handling
- Document it for future deployments

### CODE REVIEW

File: DEPLOYMENT.md [hunk 1]
```diff
@@ -10,13 +10,14 @@ If setting up a new environment `ENV_NAME` for the first time:
 - Add a new AWS configuration locally
 - Add it to `S3_BUCKET_SUFFIXES` in `.env`
 
-## 2. Deploy the contracts:
+## 2. Deploy and verify the contracts:
```
Comment (sgc-code) [inline DEPLOYMENT.md:13]: i would also update the .env.example to

```
INFURA_KEY=
ETHERSCAN_API_KEY=
ALCHEMY_KEY=
S3_BUCKET_PREFIX=argent-smartcontracts
S3_BUCKET_SUFFIXES=dev:hydrogen
S3_CONFIG_KEY=backend/config.json
DEPLOYER_GAS_PRICE=
```

plus the correct values for S3_BUCKET_SUFFIXES

it took me hours to find out this values

File: DEPLOYMENT.md [hunk 2]
```diff
@@ -10,13 +10,14 @@ If setting up a new environment `ENV_NAME` for the first time:
...
 
 ```
 ./scripts/deploy.sh --no-compile ENV_NAME 1
 ./scripts/deploy.sh --no-compile ENV_NAME 2
 ./scripts/deploy.sh --no-compile ENV_NAME 3
 ./scripts/deploy.sh --no-compile ENV_NAME 4
+./scripts/execute_script.sh --no-compile scripts/verify.js ENV_NAME
```
Comment (sgc-code) [inline DEPLOYMENT.md:22]:
```suggestion
./scripts/deploy.sh --no-compile ENV_NAME_1
./scripts/execute_script.sh --no-compile scripts/verify.js ENV_NAME_1

./scripts/deploy.sh --no-compile ENV_NAME_2
./scripts/execute_script.sh --no-compile scripts/verify.js ENV_NAME_2
```

To reinforce that verification should happen ASAP

Comment (delaaxe) [reply DEPLOYMENT.md:22]: No that's wrong, the documentation is for deploying on a single env name by running 4 different scripts

File: DEPLOYMENT.md (hunk 1 above)
Comment (sgc-code) [inline DEPLOYMENT.md:13]: i would update the .env.example with 
```
INFURA_KEY=
ETHERSCAN_API_KEY=
ALCHEMY_KEY=
S3_BUCKET_PREFIX=
S3_BUCKET_SUFFIXES=
S3_CONFIG_KEY=
DEPLOYER_GAS_PRICE=
```

And include the values for S3_BUCKET_PREFIX, S3_BUCKET_SUFFIXES and S3_CONFIG_KEY as they are not sensitive

File: DEPLOYMENT.md (hunk 2 above)
Comment (sgc-code) [reply DEPLOYMENT.md:22]: oh, so the numbers are steps??

Comment (delaaxe) [reply DEPLOYMENT.md:22]: Yea, added a sentence to make it clearer

**Review by juniset — APPROVED** [review]

**Pull request merged**